cmake/develop.cmake

```diff
@@ -0,0 +1,55 @@
+# extra
+option(BUILD_EXAMPLES "Build examples" ON)
+
+# bench test
+option(BUILD_BENCHMARK "Build benchmark" ON)
+option(BUILD_SP_BENCHMARK "Build struct_pack benchmark" OFF)
+
+# unit test
+option(BUILD_UNIT_TESTS "Build unit tests" ON)
+if (BUILD_UNIT_TESTS)
+    enable_testing()
+endif ()
+
+# coverage test
+option(COVERAGE_TEST "Build with unit test coverage" OFF)
+if (COVERAGE_TEST)
+    message(STATUS "add coverage flags")
+    if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fprofile-arcs -ftest-coverage --coverage")
+    else ()
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fprofile-instr-generate -fcoverage-mapping")
+    endif ()
+endif ()
+
+# Enable address sanitizer
+option(ENABLE_SANITIZER "Enable sanitizer(debug+gcc/clang/appleclang)" ON)
+if (ENABLE_SANITIZER)
+    if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+        if (NOT CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang|AppleClang")
+            message(FATAL_ERROR "Sanitizer supported only for Gcc/Clang/AppleClang")
+        endif ()
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=address")
```

Comment:
before use `-fsanitize=address`,  check the flag with `CHeckASan.cmake` is necessary.
due to corner case, the CI will fail.
e.g. the lib `sanitize` is not found, while the compiler work well